packages/runtime/container-runtime/src/opLifecycle/README.md

```diff
@@ -157,86 +133,7 @@ In general, compression offers a trade-off between higher compute costs, lower b
-| SeqNum: 1          | SeqNum: 2 | SeqNum: 3 | SeqNum: 4 |
-| Size: 200          | Size: 0   | Size: 0   | Size: 0   |
-| Compression: 'lz4' |           |           |           |
-+--------------------+-----------+-----------+-----------+
-```
-
-The first op in the batch is the only one with content (which is opaque due to it being compressed), the rest of the ops serve only to reserve the sequence numbers so that the state machine which rebuilds the original batch on the receiving client can reconstruct the original batch.
-
-When the batch is received by a client, it will detect the first op as being compressed, it will decompress it and store it in memory. For each empty op subsequently received, it will fetch the uncompressed content from memory and rebuild the original ops. The original ops are then processed by the runtime and applied accordingly.
-So, compression virtualizes the batch.
-
-After compression, the first op in the batch can exceed 1MB, therefore it would still be rejected. In this case, another layer of virtualization is added after compression (and before decompression, symmetrically on the receiving end).
-
-The first op in the compressed batch can be chunked into smaller ops which can be sent outside the original batch. However, to conveniently maintain the batch semantics, the last chunk (the chunk which triggers rebuilding the original op) is the first op in the new batch.
-
-To illustrate, let’s take the large batch below:
-
-```
-+--------------------+-----------+-----------+-----------+
-| Op 1               | Op 2      | Op 3      | Op 4      |
-| SeqNum: 1          | SeqNum: 2 | SeqNum: 3 | SeqNum: 4 |
-| Size: 900          | Size: 0   | Size: 0   | Size: 0   |
-| Compression: 'lz4' |           |           |           |
-+--------------------+-----------+-----------+-----------+
-```
-
-This will produce the following batches:
-
-```
-+-----------+
-| Chunk 1/3 |
-| SeqNum: 1 |
-| Size: 300 |
-+-----------+
-
-```
-
-```
-+-----------+
-| Chunk 2/3 |
-| SeqNum: 2 |
-| Size: 300 |
-+-----------+
-
-```
-
-```
-+-----------+-----------+-----------+-----------+
-| Chunk 3/3 | Op 2      | Op 3      | Op 4      |
-| SeqNum: 3 | SeqNum: 4 | SeqNum: 5 | SeqNum: 6 |
-| Size: 300 | Size: 0   | Size: 0   | Size: 0   |
-+-----------+-----------+-----------+-----------+
-```
-
-The first 2 chunks are sent in their own batches, while the last chunk is the first op in the last batch which contains the ops reserving the required sequence numbers.
-
-Notice that the sequence numbers don’t matter here, as all ops will be based off the same reference sequence number, so the sequence number will be recalculated for all, without additional work.
-
-Additionally, as compression preserves the original uncompressed batch layout in terms of the number of ops by using empty ops to reserve the sequence numbers, this ensures that the clients will always receive the exact count of ops to rebuild the uncompressed batch sequentially.
```

Comment:
I removed these two sentences, I didn't understand the point they were making.  If someone can clarify I can put something back in.

Reply:
I think it tells the reader that the last batch will have as many ops as there were in original batch. I.e. extra chunks with content do not change number of sequential ops (and thus sequence numbers) we get.

Reply:
Ok. It was confusing to me because chunking does mess with the sequence numbers - chunks 1..(n-1) get their own sequence number (but the runtime never sees those sequence numbers).

I think the concepts are covered still on the RHS of this PR, so not going to worry about it.